src/MinimalDelegation.sol

```diff
@@ -52,15 +52,24 @@ contract MinimalDelegation is IERC7821, ERC1271, EIP712, ERC4337Account, Receive
         }
     }
 
+    /// @dev This function is executeable only by the EntryPoint contract, and is the main pathway for UserOperations to be executed.
+    /// UserOperations can be executed through the execute function, but another method of authorization (ie through a passed in signature) is required.
+    /// userOp.callData is abi.encodeCall(executeUserOp.selector, (bytes32 mode, bytes executionData)) where executionData is abi.encode(Call[]).
+    function executeUserOp(PackedUserOperation calldata userOp, bytes32) external onlyEntryPoint {
+        // Parse the keyHash from the signature. This is the keyHash that has been pre-validated as the correct signer over the UserOp data
+        // and must be used to check further on-chain permissions over the call execution.
+        // TODO: Handle keyHash authorization.
+        // (bytes32 keyHash,) = userOp.signature.unwrap();
+
+        // The only valid modes for executeUserOp are BATCHED_CALL and BATCHED_CALL_CAN_REVERT.
```

Comment:
does this need to be checked?

Reply:
Hm, we could but not sure its worth. I more so meant to comment this to point out that only the EXEC_TYPE is extracted from the mode.